web3-functions/pyth/index.ts

```diff
@@ -0,0 +1,315 @@
+    const files = gistDetails.data.files;
+    if (!files)
+      throw new Error(`No files in gist`);
+    for (const file of Object.values(files)) {
+      if (file?.filename === "config.yaml" && file.content) {
+        pythConfig = YAML.parse(file.content);
+        break;
+      }
+    }
+    if (!pythConfig)
+      throw new Error(`No config.yaml loaded for PythConfig`);
+    pythConfigStorage = {
+      timestamp: Date.now() / 1e3,
+      pythConfig
+    };
+    const pythConfigStorageValue = JSON.stringify(pythConfigStorage);
+    if (pythConfig.debug) {
+      console.debug(
+        `storing fetched pythConfigStorageValue: ${pythConfigStorageValue}`
+      );
+    }
+    await storage.set("pythConfig", pythConfigStorageValue);
+  } else {
+    pythConfig = pythConfigStorage.pythConfig;
+    if (pythConfig.debug) {
+      console.debug("using pythConfig from storage");
+    }
+  }
+  return pythConfig;
+}
+async function getCurrentPrices(priceIds: string[], connection: any, debug: boolean) {
+  const latestPriceFeeds = await connection.getLatestPriceFeeds(priceIds);
+  if (latestPriceFeeds === void 0) {
+    return void 0;
+  }
+  if (debug) {
+    console.debug(`latestPriceFeeds: ${JSON.stringify(latestPriceFeeds)}`);
+  }
+  return latestPriceFeeds.map((pf: any) => {
+    return {
+      id: pf.id,
+      price: pf.getPriceUnchecked(),
+    };
+  }).filter((pf: any) => {
+    return pf !== void 0 && pf.price !== void 0;
+  }).reduce((acc: any, pf: any) => {
+    acc.set(addLeading0x(pf.id), pf.price);
+    return acc;
+  }, /* @__PURE__ */ new Map());
+}
+
+async function getLastPricesFromChain(provider: any, oracleAddress: string, priceIds: string[], ) {
+  let result = new Map();
+  let oracle;
+  for (const priceId of priceIds) {
+    oracle = new Contract(oracleAddress, ORACLE_ABI, provider);
+    const priceInfo = await oracle.getPriceUnsafe(priceId);
+    // decode result
+    const price = priceInfo[0].toString();
+    const expo = parseInt(priceInfo[2].toString());
+    const publishTime = parseInt(priceInfo[3].toString());
+    const priceInfoJson = {
+      price,
+      expo,
+      publishTime
+    }
+    result.set(priceId, priceInfoJson);
+  }
+  return result;
+}
+
+function generatePriceUpdateList(
+  priceIds: any,
+  currentPrices: any,
+  lastPrices: any,
+  allPriceIdsByItem: { [key: string]: string[] },
+  validTimePeriodSeconds: number,
+  deviationThresholdBps: number,
+  debug: boolean
+): string[] {
+
+  const priceUpdateList: string[] = [];
+
+  for (const item of Object.keys(priceIds)) {
+    let composedLastPrice = 0;
+    let composedCurrentPrice = 0;
+    const priceIdObj = priceIds[item];
+    for (const priceFeed of priceIdObj) {
+      const priceId = addLeading0x(priceFeed.id);
+      const currentPrice = currentPrices.get(priceId);
+      const lastPrice = lastPrices.get(priceId);
+
+      if (currentPrice.publishTime - lastPrice.publishTime > validTimePeriodSeconds) {
```

Comment:
if this happen and we don't break, we may add twice the feed.

Reply:
So let's break but then set composedLastPrice and composedCurrentPrice to 1 to not break the code to check the diff

Reply:
ok my bad you use set at the end